src/zh/hanmanwuzhe/src/eu/kanade/tachiyomi/extension/zh/hanmanwuzhe/hanmanwuzhe.kt

```diff
@@ -0,0 +1,233 @@
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            setUrlWithoutDomain(element.select("div.neirong > a.name, a.img").attr("href"))
+            title = element.select("div.neirong > a.name, a.txt").text()
+            thumbnail_url = element.select("div.pic > a > img, a.img > img").imgAttr()
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val genres = filters.filterIsInstance<GenreListFilter>().first()
+            .state.filter { it.state }.map { it.id }
+
+        val url = if (query.isNotBlank() || genres.isEmpty()) {
+            "$baseUrl/index.php/search"
+                .toHttpUrl().newBuilder()
+                .addQueryParameter("key", query)
+                .toString()
+        } else {
+            "$baseUrl/category/${genres.joinToString(",")}" +
+                if (page > 1) {
+                    "/page/$page/"
+                } else {
+                    ""
+                }
+        }
+
+        return GET(url, headers)
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.select("h1 > div.title").text()
+            author = document.select("div.info > p.tage:contains(作者：)").text().substringAfter("作者：")
+            genre = document.select("p.tage:contains(类型：) > a[href*=/category/tags/]").eachText().joinToString()
+            thumbnail_url = document.select("div.info > div.img > img").imgAttr()
+            description = document.select("div.text").text()
+            status = SManga.COMPLETED
+            update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        }
+    }
+
+    override fun chapterListSelector() = "div.listbox > ul.list li"
+
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        url = element.select("a").attr("href")
+        name = element.select("a").first()!!.ownText()
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        println(document.select(chapterListSelector()))
+        return document.select(chapterListSelector()).map { chapterFromElement(it) }.reversed()
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select("div.chapterbox li.pic img").mapIndexed { idx, img ->
+            Page(idx, imageUrl = img.imgAttr())
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
+
+    override fun imageRequest(page: Page): Request {
+        val newHeaders = headersBuilder()
+            .set("Accept", "image/avif,image/webp,image/png,image/jpeg,*/*")
+            .build()
+
+        return GET(page.imageUrl!!, newHeaders)
+    }
+
+    private class Genre(name: String, val id: String = name) : Filter.CheckBox(name)
```

Comment:
I have a question, why use CheckBox if only one genre can be chosen?

Reply:
I don't really know the functions, sorry 😅, is there a function for a radio button?

Reply:
You can use Filter.Select to create drop-down list.

Reply:
is this alright?

Reply:
Emmm, I would suggest using Pair so you can put genre name and url in the same array. You can read code of other extensions, like [manhuaren](https://github.com/keiyoushi/extensions-source/blob/83a2c48ef5f4ec5162c2b60541c9adacd0421f8b/src/zh/manhuaren/src/eu/kanade/tachiyomi/extension/zh/manhuaren/Manhuaren.kt#L590C6-L590C47) or [zerobyw](https://github.com/keiyoushi/extensions-source/blob/83a2c48ef5f4ec5162c2b60541c9adacd0421f8b/src/zh/zerobyw/src/eu/kanade/tachiyomi/extension/zh/zerobyw/Zerobyw.kt#L153)